kytos/core/events.py

```diff
@@ -1,42 +1,30 @@
 """Module with Kytos Events."""
 import json
+from dataclasses import dataclass, field
 from datetime import datetime
-from uuid import uuid4
+from uuid import UUID, uuid4
 
 from kytos.core.helpers import now
 
 
+@dataclass
 class KytosEvent:
     """Base Event class.
 
     The event data will be passed in the `content` attribute, which should be a
     dictionary.
     """
 
-    def __init__(self, name=None, content=None, trace_parent=None, priority=0):
-        """Create an event to be published.
+    name: str
```

Comment:
It's nice to have `KytosEvent` decorated as a `@dataclass`. However, we've lost useful context from the docstring, for instance down the road, someone trying to read or understand what `trace_parent` is for, or trying to understand if `priority` if it's ascending or descending. Can you document them again? No need to keep the old types in the docstrings though since actual type hints are far superior.

Reply:
Also, if we'll improve here and evolve with type hints, let's also take the opportunity to use `pydantic` [dataclass decorator](https://docs.pydantic.dev/latest/usage/dataclasses/), just so if a NApp incorrectly passes the argument types then it raises a validation error at the call site. It's supposed to provide similar funcionality according to `pydantic` docs. In that case it'd be great to grep over the other repos to ensure that `None` isn't being set as `name`, which it shouldn't but maybe there might be an old unit test as you've found here.

What do you think?

Reply:
Added back in + spruced up the old docstrings. Also switched to the pydantic dataclass as suggested.